src/librarySystem/Library.java

```diff
@@ -57,10 +57,8 @@ public long returnBook(LocalDate currentDay, PrintedProduction book, Customer cu
         }
         return countFine;
     }
-
     public List<PrintedProduction> filter(Filter filter) {
-        PrintedProduction[] filteredBooks = filter.filter(books.values().toArray(new PrintedProduction[0]));
-        return Arrays.asList(filteredBooks);
+        return filter.filter((List<PrintedProduction>) books.values());
```

Comment:
Note: It will fail if the `books` map will return not a `List` but different kind of `Collection`. Looks ok for now.